### PR TITLE
Formatting email

### PR DESCRIPTION
**Pull request by volodymyrss**
* "scientific" email text
* refactoring of email functionality
* tests with reusable functions

will use instead of https://github.com/oda-hub/dispatcher-app/pull/133 since can not assign reviewers

### CODE REVIEW

File: cdci_data_analysis/analysis/email_helper.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    finally:
+        if server:
+            server.quit()
+
+    store_email_info(message, status, scratch_dir)
+
+    return message
+
+
+def store_email_info(message, status, scratch_dir):
```
Comment (burnout87) [inline cdci_data_analysis/analysis/email_helper.py:158]: This is still at the old version. I had slightly changed the way the email file name was formatted. Same for the checks

**Review by burnout87 — COMMENTED** [review]
I can adapt with the latest changes I had done

**Comment by ferrigno** [comment]
I received the submission email. Nice work. I have just a few suggestions between stars below ** :

Dear User,

[...]

You can inspect the status of your job using the 
*this url* or repeating the request (either with frontend or with *the python API query at the end of this message.*

In case you have questions or issues you can contact us at this email.

Kind regards,

The ODA team.

*Python API code:*

File: cdci_data_analysis/analysis/email_helper.py (hunk 1 above)
Comment (burnout87) [reply cdci_data_analysis/analysis/email_helper.py:158]: I can also push those changes

Comment (volodymyrss) [reply cdci_data_analysis/analysis/email_helper.py:158]: The refactoring changes and so have diverged with the master so it became difficult to merge. Not good the PR stayed or a whole while development progressed. 
So let's quickly get over this one?

You will make a simple change, yes?

Comment (burnout87) [reply cdci_data_analysis/analysis/email_helper.py:158]: exacly

**Comment by ferrigno** [comment]
I see that the URL points to the production instance, so I submitted the job to production...my bad.

One suggestions to the ready email, as before.

 Dear User,

[...]

*You can inspect the result at this url and click to the submit button or excecuting the 
python API code at the end of this emal.*


In case you have questions or issues you can contact us at this email.

Kind regards,

The ODA team.

*Python API code:*

**Comment by volodymyrss** [comment]
> I see that the URL points to the production instance, so I submitted the job to production...my bad.
> 
> One suggestions to the ready email, as before.

Ok, I implemented them.

You can also make PRs directly in browser, by finding file:

https://github.com/oda-hub/dispatcher-app/blob/formatting_email/cdci_data_analysis/flask_app/templates/email.html

and clicking edit button, as so:

https://github.com/oda-hub/dispatcher-app/pull/144

Then the change propagates faster, more transparently, less work for all.

**Comment by ferrigno** [comment]
Ok, thanks for the info. I can implement future cosmetic changes by myself, knowing where I can do it.

**Review by burnout87 — COMMENTED** [review]
I just implemented those few changes as agreed.

**Comment by volodymyrss** [comment]
> I just implemented those few changes as agreed.
> 
> I cannot ask for your review since you created the PR in first place.

yeah that's why I created it, and probably better not too switch people and make shorter PRs :)

**Comment by volodymyrss** [comment]
> > I just implemented those few changes as agreed.
> > I cannot ask for your review since you created the PR in first place.
> 
> yeah that's why I created it, and probably better not too switch people and make shorter PRs :)

If all is good for you, please approve, I merge, and we continue in other PRs as needed.

**Review by burnout87 — APPROVED** [review]

**Pull request merged**